rero_ils/modules/items/api/circulation.py

```diff
@@ -793,15 +793,34 @@ def prior_checkout_actions(self, action_params):
             ):
                 item, receive_actions = self.receive(**action_params)
                 actions.update(receive_actions)
-            if loan['state'] == LoanState.ITEM_IN_TRANSIT_TO_HOUSE:
-                item, cancel_actions = self.cancel_loan(pid=loan.get('pid'))
+            elif loan['state'] == LoanState.ITEM_IN_TRANSIT_TO_HOUSE:
+                item, cancel_actions = self.cancel_loan(**action_params)
                 actions.update(cancel_actions)
                 del action_params['pid']
         else:
             loan = get_loan_for_item(item_pid_to_object(self.pid))
             if (loan and loan['state'] != LoanState.ITEM_AT_DESK):
                 item, cancel_actions = self.cancel_loan(pid=loan.get('pid'))
                 actions.update(cancel_actions)
+        # CHECKOUT_1_2_2: checkout denied if some pending loan are linked to it
+        # with different patrons
+        # Except while coming from an ITEM_IN_TRANSIT_TO_HOUSE loan because
+        # the loan is cancelled and then came up in ON_SHELF to be checkout
+        # by the second patron.
+        if self.status == ItemStatus.ON_SHELF and \
```

Comment:
I think here, you do not need to check:
`loan['state'] != LoanState.ITEM_IN_TRANSIT_TO_HOUSE` because you will never have an item `in_transit` and `on_shelf` at same time.

If I were you, I replace it with `loan['state'] == LoanState.PENDING`

Reply:
During the process of a checkout of a ITEM_IN_TRANSIT_TO_HOUSE, a loan is canceled. And the ITEM become ON_SHELF. That's why I add this condition.
I will try what you suggest and will come back to you about the result.

Reply:
Tests CHECKOUT_1_2_2 didn't pass with `loan['state'] == LoanState.PENDING` :disappointed:

Reply:
I got it. 
if the loan is cancelled, then you do not need to check its state at all:
`if self.status == ItemStatus.ON_SHELF:`
and then you continue to compare  the  `pending.patron_pid` with `loan.pid`
we can take a look at it together if you want

Reply:
After the loan is cancelled, the variable is not updated as it becomes `None`. And then the checkout cannot pass. And so the item didn't becomes ON_LOAN.
I will add a comment `TODO` to check this problem later.